sycl/doc/extensions/proposed/sycl_ext_oneapi_async_memory_alloc.asciidoc

```diff
@@ -0,0 +1,729 @@
+is allocated and freed from.
+
+The asynchornous memory allocation and free are exposed via new free functions
+taking a queue, which enqueue an asynchronous malloc or free command and return
+immediately. The functions can take a memory pool object in order to allocate
+from that memory pool specifically, however, if no memory pool object is
+provided there is a default memory pool which will be used.
+
+The memory pool introduced is a dynamic memory pool, as such no memory is
+immediately allocated on construction, instead memory is allocated by the OS
+for the pool when allocations are made and are freed back to the OS when the
+queue which enqueued the malloc and free commands is synchronized with.
+
+The immediate benefit of using the asynchornous malloc and free functions, as
+opposed to the existing synchronous malloc and free functions, is that they no
+longer block the calling thread. So applications which wish to allocate
+temporary USM memory but not carry around the pointer no longer have to
+synchronize with the execution of kernels accessing that memory and call the
+free function before continuing.
+
+A further benefit comes from the use of a memory pool for allocations, as
+malloc and free commands are asynchronous, the runtime can opportunistically
+re-use memory allocations from one kernel to another, without any intermediate
+freeing and re-allocation. This benefit can be further extended beyond queue
+synchronization by specify a threshold, which will instruct the runtime to (if
+possible) maintain a certain size of memory and not release that back to the OS
+even when the queue is synchronized with.
+
+There are also other properties which can be used when constructing a memory
+pool object to control other aspects of how the memory is allocated.
+
+== Integration with other extensions
+
+It is a priority for this extension to integrate with the
+link:../supported/sycl_ext_oneapi_graph.asciidoc[
+sycl_ext_oneapi_graph] extension. The asynchornous commands introduces in this
+extension are intended to be compatible with the SYCL Graph record and replay
+mode, however, the specifics of this will be defined in that extension.
+
+It is a priority for this extension to integrate with the
+link:../supported/sycl_ext_oneapi_bindless_images.asciidoc[
+sycl_ext_oneapi_bindless_images] extension. It is the intention that the memory
+allocations from the memory pool in this extension be used for the bindless
+images, however, this will need to be investigated further and so the details of
+this are not included in this version of the extension.
+
+== Examples
+
+There are various ways in which this extension can be used but a typical usage
+of the memory pool and the asynchronous malloc and free commands is described
+below. In this example an explicit memory pool is created and this is used to
+share memory allocated from the OS between multiple asynchornous malloc
```

Comment:
asynchronous

Reply:
Fixed in https://github.com/intel/llvm/pull/14800/commits/2c51adf28ece21c0e91c2c0ec85440ae1153e536.